src/scheduler/fifo.c

```diff
@@ -798,6 +745,42 @@ scheduling_cycle(int sd, char *jobid)
 	return 0;
 }
 
+/**
+ * @brief
+ * 	check whether given server send any priority command or not
+ *
+ * @param[in] sconn - pointer to server struct
+ *
+ * @return int
+ * @retval 1 - has priority command
+ * @retval 0 - no priority command
+ */
+static int
+has_priority_cmd(svr_t *sconn)
+{
+	svr_t sconn_copy;
+
+	if (sconn->secondary_fd == -1)
+		return 0;
+
+	/* Make copy of given server struct as we don't want to override original command */
+	sconn_copy.primary_fd = sconn->primary_fd;
+	sconn_copy.secondary_fd = sconn->secondary_fd;
+	sconn_copy.cmd.cmd = -1;
+	sconn_copy.cmd.jid = NULL;
+	CLEAR_LINK(sconn_copy.all_svrs_link);
+
+	if (!get_sched_cmd_noblk(&sconn_copy))
+		return 0;
+
+	if (sconn_copy.cmd.cmd == SCH_SCHEDULE_RESTART_CYCLE) {
+		if (sconn_copy.cmd.jid)
+			free(sconn_copy.cmd.jid);
+		return 1;
```

Comment:
Well my idea is to change only this func in case we ever add another high priority command so we don't have to sched main_sched_loop()

Reply:
How is knowing there is a high priority command useful without knowing the command?  Right now there is only one, so returning 1/0 works.  Once there are two, you won't know what the high priority command is, so you would have to change main_sched_loop(), because you wouldn't want to restart the cycle if the second command is sent.

Reply:
removed this func and added this code in main_sched_loop()... this comment is invalid now.